### PR TITLE
Expanded installation section

### PR DESCRIPTION
**Pull request by Paulocracy**
I added an installation pathway for CPLEX in connection with CNApy, and tried to add the missing figure. As long as I didn't miss any simple way to do it, installing CPLEX seems to be quite tedious for a normal user :-| Should be automate this process somehow?

### CODE REVIEW

File: content/03.installation.md [hunk 1]
```diff
@@ -1,6 +1,61 @@
...
+In order to use "old" function from the original Matlab/Octave *CellNetAnalyzer*, you need one of the following programs installed at your systems (follow the links to the programs for download and installation instructions)...
+1. [MATLAB](https://www.mathworks.de/products/matlab/index.html), **in a version from 2019b up to 2021b (older and newer versions are not supported!)**.
+2. [GNU Octave](https://www.gnu.org/software/octave/index) installed at your system, **in a version >6.4.0 (older versions are not supported!)**.
+
+...as well as the original [*CellNetAnalyzer*](http://www2.mpi-magdeburg.mpg.de/projects/cna/cna.html/).
+
+
+### Additional solvers (optional)
+
+*Note: Additional solvers are only possible if you've installed CNApy according to the first main method, i.e. by using conda*
```
Comment (VonAlphaBisZulu) [inline content/03.installation.md]: This should be possible in any case (installation via conda, via Windows installer and manually (i.e. cnapy-dev)). See below.

File: content/03.installation.md [hunk 2]
```diff
@@ -1,6 +1,61 @@
...
+
+CNApy itself is already packaged and installed with the open source linear programming solver [GLPK](https://www.gnu.org/software/glpk/) which is fast enough for simple calculations.
+However, for large models and more complications calculations (such as Minimal Cut Set analyses), it is recommended to install a more sophisticated and faster solver such as [CPLEX](https://www.ibm.com/products/ilog-cplex-optimization-studio) or [Gurobi](https://www.gurobi.com/). Both of these solvers have free academic license programs.
+
+#### Connect CPLEX to CNApy
+
+*Note: This connection is only possible if you've installed CNApy according to the first main method, i.e. by using conda*
```
Comment (VonAlphaBisZulu) [inline content/03.installation.md]: This should be possible in any case (installation via conda, via Windows installer and manually (i.e. cnapy-dev)). See below.

File: content/03.installation.md [hunk 3]
```diff
@@ -1,6 +1,61 @@
...
+
+Once installed CPLEX on your system (as described on [CPLEX's website](https://www.ibm.com/products/ilog-cplex-optimization-studio)), use conda in order to activate CNApy's conda environment:
+
+```bash
+conda activate cnapy-dev
+```
+
+Once you're inside the cnapy-dev environment, install the CPLEX package by using pip as follows:
```
Comment (VonAlphaBisZulu) [inline content/03.installation.md]:
```suggestion
Once you activated the CNApy environment, navigate to your CPLEX installation (e.g. via ```cd "C:\Program Files\IBM\ILOG\CPLEX_Studio221"```) and install the CPLEX package as follows:
```

File: content/03.installation.md [hunk 4]
```diff
@@ -1,6 +1,61 @@
...
+*Note: Additional solvers are only possible if you've installed CNApy according to the first main method, i.e. by using conda*
+
+CNApy itself is already packaged and installed with the open source linear programming solver [GLPK](https://www.gnu.org/software/glpk/) which is fast enough for simple calculations.
+However, for large models and more complications calculations (such as Minimal Cut Set analyses), it is recommended to install a more sophisticated and faster solver such as [CPLEX](https://www.ibm.com/products/ilog-cplex-optimization-studio) or [Gurobi](https://www.gurobi.com/). Both of these solvers have free academic license programs.
+
+#### Connect CPLEX to CNApy
+
+*Note: This connection is only possible if you've installed CNApy according to the first main method, i.e. by using conda*
+
+Once installed CPLEX on your system (as described on [CPLEX's website](https://www.ibm.com/products/ilog-cplex-optimization-studio)), use conda in order to activate CNApy's conda environment:
```
Comment (VonAlphaBisZulu) [inline content/03.installation.md]:
```suggestion
Once installed CPLEX on your system (as described on [CPLEX's website](https://www.ibm.com/products/ilog-cplex-optimization-studio)), the API of your CPLEX installation must be connected to your CNApy's conda environment. Therefore, first open the command line tool and activate your CNApy's conda environment. Depending on your installation method, the name (or path) to your conda installation varies.
```

File: content/03.installation.md [hunk 5]
```diff
@@ -1,6 +1,61 @@
...
+
+```bash
+conda activate cnapy-dev
+```
+
```
Comment (VonAlphaBisZulu) [inline content/03.installation.md:48]:
```suggestion

or

```bash
conda activate cnapy-X.X.X```

or

```bash
conda activate C:\Users\YourUserName\cnapy```
```

File: content/03.installation.md [hunk 6]
```diff
@@ -1,6 +1,61 @@
...
+Once you're inside the cnapy-dev environment, install the CPLEX package by using pip as follows:
+
+```bash
+(cnapy-dev) pip install cplex
```
Comment (VonAlphaBisZulu) [inline content/03.installation.md:52]:
```suggestion
```
Correct me if I'm wrong, but this step and also the following steps should not be required, (apart from setup.py install)

File: content/03.installation.md [hunk 7]
```diff
@@ -1,6 +1,61 @@
...
+Now, go to the folder where you installed CPLEX:
+
+```bash
+(cnapy-dev) cd your_path_to_cplex
+```
+
+Then, install CPLEX's Python API by using the following command (for further details, see [this](https://www.ibm.com/docs/en/icos/20.1.0?topic=cplex-setting-up-python-api)):
+
+```bash
+(cnapy-dev) python setup.py install
```
Comment (VonAlphaBisZulu) [inline content/03.installation.md]:
```suggestion
(cnapy) python ./python/setup.py install
```

File: content/03.installation.md [hunk 8]
```diff
@@ -1,6 +1,61 @@
...
+```
+
+Now add an environmental variable called PYTHONPATH which leads the cplex Python package to CPLEX. Under Linux, you can do this as follows:
+
+```bash
+(cnapy-dev) export PYTHONPATH=your_path_to_cplex/python/your_cplex_version/your_system
+```
+
+Under Windows, you can do this by searching for your system's "environmental variables" and adding the variable PYTHONNOUSERSITE with the value "your_path_to_cplex/python/your_cplex_version/your_system" using Windows's environmental variables setting window.
```
Comment (VonAlphaBisZulu) [inline content/03.installation.md:75]:
```suggestion
```
Also not sure if this is required...

**Review by VonAlphaBisZulu — CHANGES_REQUESTED** [review]
Hey, I left you some comments. Not really sure, if I'm using this comment tool correctly

File: content/03.installation.md (hunk 4 above)
Comment (sthiele) [reply content/03.installation.md]: I would rephrase it like this: 
Once CPLEX is installed  on your system (as described on [CPLEX's website](https://www.ibm.com/products/ilog-cplex-optimization-studio)), the CPLEX Python API must be installed in the CNApy conda environment.

Also instead of: your CNApy's conda environment 
I would write: the CNApy conda environment

**Comment by axelvonkamp** [comment]
For CPLEX (and Gurobi) it may perhaps be clearer to distinguish between the free demo version (which is OK for small models) that can be installed with pip without any further configuration and the full version which needs more configuration steps.

File: content/03.installation.md (hunk 8 above)
Comment (Paulocracy) [reply content/03.installation.md:75]: This is actually required, because otherwise, the CPLEX Community Edition is used :-(

File: content/03.installation.md (hunk 6 above)
Comment (Paulocracy) [reply content/03.installation.md:52]: I think they are necessary because cnapy-dev does not contain these packages by default, at least I had to install them :3

File: content/03.installation.md (hunk 8 above)
Comment (VonAlphaBisZulu) [reply content/03.installation.md:75]: Yes, but only because you installed the community edition via pip before. If you skip this step and only install the API from the full CPLEX installation you should be fine

File: content/03.installation.md (hunk 6 above)
Comment (VonAlphaBisZulu) [reply content/03.installation.md:52]: There should be a way to only install dependencies. Maybe uninstall cplex community again with pip, but keep the dependencies?

**Comment by Paulocracy** [comment]
I just uploaded a commit which tries to merge all of your ideas, feel free to further comment it :-)
Does anyone of you have a successfull Gurobi connection to CNApy? Since I do not have it installed, I cannot deduce the installation instructions for it.

**Comment by VonAlphaBisZulu** [comment]
Yes, gurobi works for me. The procedure is the same as for CPLEX. Gurobi's path to setup.py is in Windows:
C:\gurobi951\win64\setup.py

Again, there is a community version available from pip. People might be confused why Python uses the community version while they have the full Gurobi optimizer installed.

**Comment by Paulocracy** [comment]
> Yes, gurobi works for me. The procedure is the same as for CPLEX. Gurobi's path to setup.py is in Windows: C:\gurobi951\win64\setup.py
> 
> Again, there is a community version available from pip. People might be confused why Python uses the community version while they have the full Gurobi optimizer installed.

Great :-) So how do we tell CNApy which Gurobi it should use? I.e., what is Gurobi's analogy to PYTHONPATH?

**Comment by VonAlphaBisZulu** [comment]
I would say, the same as for CPLEX, just don't install the community version via pip. Or uninstall the community version (but not its dependencies) before installing the pro version. You'll never need the community and the full version installed at the same time in your environment anyways.

**Comment by axelvonkamp** [comment]
Octave works with version 5.2 or higher. Also mention that the presence of an older octave version should be avaoided because this can mess with CNApy startup...

**Comment by Paulocracy** [comment]
I updated the Octave explanation and added a first Gurobi installation explanation, however, since I'm not sure how to set up the "real" Gurobi instead of the demo version for optlang, I left a !!! which indicates that someone with knowledge about that should add the instructions :-)

File: content/03.installation.md (hunk 6 above)
Comment (Paulocracy) [reply content/03.installation.md:52]: The "cplex" package contains the CPLEX Community Edition but just as a "dummy" for the full version. Hence, uninstalling cplex wouldn't help. Instead, we have to tell cplex where the full CPLEX is.

**Comment by Paulocracy** [comment]
Thanks to the new installation wizards in CNApy 1.0.9, the installation process is now much simpler and I uploaded an associated updating PR (#4). Therefore, all changes here are not neccessary anymore.
